### PR TITLE
[RFR] Feat two players

### PR DESCRIPTION
**Pull request by mchaffotte**
TwoPlayers

### CODE REVIEW

File: src/game.py [hunk 1]
```diff
@@ -1,48 +1,76 @@
...
-        console.display("** You won **")
+    if state == State.VICTORY:
+        display(new_pawns, new_fences)
+        console.display("\n *** Player # " + str(player_turn) + " won ***\n")
 
 
-def display(pawn, fences):
+def get_next_player(player_turn, pawns):
+    if player_turn + 1 > len(pawns):
+        player_turn = 1
```
Comment (Kmaschta) [inline src/game.py]: Argument is directly updated. This might cause a side effect.

File: src/game.py [hunk 2]
```diff
@@ -1,48 +1,76 @@
...
-    board = get_board(pawn, fences)
+    board = get_board(pawns, fences)
     console.display_game(board)
 
 
 def is_a_victory(pawn):
-    return pawn.x == BASE_LINE_SIZE - 1
+    victory: None
+    if pawn.goal == Orientation.EAST:
+        victory = pawn.x == 0
```
Comment (Kmaschta) [inline src/game.py:65]: Argument is directly updated. This might cause a side effect.

File: src/console.py [hunk 3]
```diff
@@ -50,8 +52,9 @@ def get_top_border(board):
     return border
 
 
-def prompt_action():
-    key = prompt("Where do you want to move the pawn?\nChoose in list and type ENTER " + get_action_list())
+def prompt_action(player_number):
+    print("Possible actions: " + get_action_list())
+    key = prompt("Player #" + str(player_number) + ": Where do you want to move the pawn? ")
```
Comment (Kmaschta) [inline src/console.py]: fstrings! :p
(not mandatory)

File: src/game.py [hunk 4]
```diff
@@ -1,48 +1,76 @@
...
 def init_game():
     center = math.floor(BASE_LINE_SIZE / 2)
-    return Pawn(0, center)
+    return [Pawn(0, center, Orientation.WEST), Pawn(BASE_LINE_SIZE - 1, center, Orientation.EAST)]
 
 
-def progress(pawn):
-    new_pawn = pawn
+def progress(pawns):
+    new_pawns = pawns
```
Comment (Kmaschta) [inline src/game.py]: Be aware: you didn't copied the value of your pawns, you just stored the reference of your pawns.
So, if you change new_pawns later, pawns will be changed too

File: src/game.py [hunk 5]
```diff
@@ -1,48 +1,76 @@
...
+        display(new_pawns, new_fences)
+        action = console.prompt_action(player_turn)
         if action == Action.EXIT:
-            quit = True
+            state = State.QUIT
         else:
             try:
-                new_pawn = act(action, new_pawn, new_fences)
+                new_pawn = act(action, new_pawns[player_turn - 1], new_fences)
+                new_pawns = deepcopy(new_pawns)
```
Comment (Kmaschta) [inline src/game.py:38]: I'd recommend to deepcopy `pawns` at start, and then update new_pawns as you need

File: src/board.py [hunk 6]
```diff
@@ -31,26 +32,28 @@ def build():
...
-            if pawn.x == x and pawn.y == y:
-                line.append(Item.PAWN)
-            elif is_odd(x) or is_odd(y):
-                line.append(Item.NO_FENCE)
-            else:
-                line.append(Item.SQUARE)
-        board.append(line)
-    return board
+def get_pawn_item(index):
+    item = None
```
Comment (ThieryMichel) [inline src/board.py]:
```py
return Item.PAWN_1 if index == 0 else Item.PAWN_2
```

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/game.py [hunk 7]
```diff
@@ -50,10 +51,8 @@ def progress(pawns):
 
 def get_next_player(player_turn, pawns):
     if player_turn + 1 > len(pawns):
```
Comment (ThieryMichel) [inline src/game.py]: return 1 if player_turn + 1 > len(pawns) else player_turn + 1

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

**Pull request merged**